pkg/aws/ec2/api/helper.go

```diff
@@ -208,7 +209,7 @@ func (h *ec2APIHelper) DeleteNetworkInterface(interfaceId *string) error {
 	}
 
 	err := retry.OnError(defaultBackOff, func(err error) bool { return true }, func() error {
-		_, err := h.ec2Wrapper.DeleteNetworkInterface(deleteNetworkInterface)
+		_, err := h.ec2Wrapper.DeleteNetworkInterface(context.TODO(), deleteNetworkInterface)
```

Comment:
Maybe we want to use other than TODO()?

Reply:
this part of API does not use context. I am adding TODO here as the interface it calls to has parameter for context. We can make our EC2 API calls to have context parameter, which provides other module to use it. But that change is related to other PR.